### DISCUSSION TITLE
TIVORemotePF.html works with 3.8f but not  with 4.0

### DISCUSSION THREAD

**Discussion by f520k510** [GerryDazoo/Slinger#537] [General]
[TIVOremotePF.zip](https://github.com/GerryDazoo/Slinger/files/10774367/TIVOremotePF.zip)

I don't want to sound like I am asking others to debug my remote file, but the remote files that I modified the channel list in is working fine on 3.8f on my raspberry pi but is not working on my windows running python 3.9 and slinger 4.0, nor with the slingbox_server.exe.  It says that error loading definition of tivoremotepf.html.

How can I debug a html remote file?
I open the html remote with chrome and it display the buttons just fine, and the channel list pops up just fine.
I began to wonder if the channel list is too long, which has around 148 channels, for flask module to recognize.
Thanks.

**Comment by GerryDazoo** [comment]
Where is this " error loading definition of tivoremotepf.html." coming from? Please supply logs.
Don't forget file names on Linux are  case senvitive

**Comment by f520k510** [comment]
here is the log from my windows:


Version : 4.0 Running on Windows-10-10.0.22621-SP0 pid= 14444 D:\Slinger-main\Slinger-main\slingbox_server.exe
Using config file config.ini
Connection Manager Running on port 65432 with 3 max streams using URL nadalliu.
BOXES [('sb1', '700a'), ('sb2', '700b'), ('sb3', '4k')]
BOX URL /nadalliu/700a
Building page for 700a
Reading Custom Remote definition from TIVORemotePF.html
Error reading remote definition file, using defaults 'cp932' codec can't decode byte 0x86 in position 8542: illegal multibyte sequence
BOX URL /nadalliu/700b
Building page for 700b
Reading Custom Remote definition from 700a.txt
Error reading remote definition file, using defaults [Errno 2] No such file or directory: '700a.txt'
Streamer Running:  3 config.ini 700a 700a 65432 8388608
BOX URL /nadalliu/4k
Building page for 4k
WARNING: Remote Control Code not set. Set RemoteCode or VideoSource
Defaulting to 0 which is probably wrong
Reading Custom Remote definition from TIVORemotePF.html
Streamer Running:  3 config.ini 700b 700b 65432 8388608
700a Checking for slingbox at ('192.168.1.21', 5216) 0
700b Checking for slingbox at ('192.168.1.21', 5216) 0
Error reading remote definition file, using defaults 'cp932' codec can't decode byte 0x86 in position 8542: illegal multibyte sequence
Streamer Running:  3 config.ini 4k 4k 65432 8388608
700a ('192.168.1.21', 5216) OK
700b ('192.168.1.21', 5216) OK
starting up on  port 65432 8388608
700a Using slingbox at  ('192.168.1.21', 5216)
700b Using slingbox at  ('192.168.1.21', 5216)
4k Checking for slingbox at ('192.168.1.25', 5211) 0
Streamer:  700a Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Streamer:  700b Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
4k ('192.168.1.25', 5211) OK
4k Using slingbox at  ('192.168.1.25', 5211)
Streamer:  4k Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
 * Serving Flask app 'slingbox_server'
 * Debug mode: off
Broadcast Responder Running
Server IP address is 192.168.1.8

I updated slingbox_server.py to 4.0 on my raspberry pi and it is still working.
Must be something specific to windows.

**Reply by GerryDazoo** [reply]
Did you go to character 8542 to see what it might be complaining about?

**Reply by GerryDazoo** [reply]
I'm seeing Some Unicode stuff
"text":"518 フェニックステレビ（鳳凰衛視）"  Is this expected?

**Reply by f520k510** [reply]
Yes, they are Japanese characters.
I think it doesn't like the Japanese characters.

**Comment by GerryDazoo** [comment]
![image](https://user-images.githubusercontent.com/102599361/219870883-f7a5882f-3785-422a-a679-1e705def6cce.png)
Working for me.  Windows/Python. Slinger 4.0

**Reply by GerryDazoo** [reply]
I can't work on this right now. Might be an issue with the Windows exe. I'll get back to it ASAP.

**Reply by f520k510** [reply]
Thanks and take your time.
I just don't understand why it works for you on windows.

**Reply by GerryDazoo** [reply]
I thinks it's because I'm running python and not the exe. Standby

**Reply by GerryDazoo** [reply]
Very strange. I've got 4 difference pythons installed on various PCs on my network that I use. Two of them load the remote correctly and 2 of them don't and I have no idea why. For sure it's dying on the Japanese characters. For a test I deleted the offending characters and retried. It did die a little later but it did get past some other Japanese characters before failing.
Unfortunately, the python version I'm using to generate the exe is one of the ones failing.
Just wondering how you generated this file. In particular what encoding did you use for the Japanese characters? Two of my pythons don't think it's UTF-8.

**Reply by GerryDazoo** [reply]
Give this a try.
[slingbox_server.zip](https://github.com/GerryDazoo/Slinger/files/10774952/slingbox_server.zip)

**Reply by f520k510** [reply]
The exe file is working fine for me.
So I assume it has to do with which version of python we are using.
On my windows I run python 3.9, which has issues with these Japanese characters.

**Reply by f520k510** [reply]
> Very strange. I've got 4 difference pythons installed on various PCs on my network that I use. Two of them load the remote correctly and 2 of them don't and I have no idea why. For sure it's dying on the Japanese characters. For a test I deleted the offending characters and retried. It did die a little later but it did get past some other Japanese characters before failing.
> Unfortunately, the python version I'm using to generate the exe is one of the ones failing.
> Just wondering how you generated this file. In particular what encoding did you use for the Japanese characters? Two of my pythons don't think it's UTF-8.

I think I used notepad on  win10 to edit the html file.
I copied and pasted the channel names from a Japanese web site, if that matters.

**Reply by GerryDazoo** [reply]
Did you try, the exe I posted. It's shouldn't fail loading the remote file.

**Reply by f520k510** [reply]
Yes, I did.  As I said a few posts above it works fine.

**Comment by pfuller88** [comment]
Mine works fine on 4, python version on a Pi, although I'm using the mapped remote picture version.   I'll try the above version later today and reply.  I've also expanded my version to toggle both the mapped remote image and the non-mapped view into one remote.  try to upload later  today.

PF

**Reply by pfuller88** [reply]
Here is the latest html that I use,.  Works for me on a raspberry Pi on version 4.   There is a basic readme file in the zip file.
[TIVOremotePFv3.zip](https://github.com/GerryDazoo/Slinger/files/10775073/TIVOremotePFv3.zip)
